strawberryfields/ops.py

```diff
@@ -640,27 +613,24 @@ class Channel(Transformation):
     """
 
     def merge(self, other):
-        # check that other is an identical channel, and that there are
-        # no extra dependencies <=> no RegRefTransform parameters
-        if isinstance(other, self.__class__) \
-                and len(self._extra_deps)+len(other._extra_deps) == 0:
-            # check that all other parameters are identical
-            if np.all(self.p[1:] != other.p[1:]):
-                raise MergeFailure('Other parameters differ.')
-
-            # determine the new loss parameter
-            T = self.p[0] * other.p[0]
+        if not self.__class__ == other.__class__:
+            raise MergeFailure('Not the same channel family.')
 
+        # channels can be merged if they are the same class and share all the other parameters
+        if self.p[1:] == other.p[1:]:
```

Comment:
I suppose at one stage, we might consider merging gates which differ in a parameter other than the first?

Reply:
Maybe, but there's no general rule for it. `Gate` subclasses are guaranteed to be one-parameter subgroups w.r.t. `p[0]` but nothing more, and for `Channel` subclasses we have not even properly defined how `p[0]` should be treated...